src/components/Traceroute/Traceroute.tsx

```diff
@@ -55,89 +43,206 @@ const TracerouteTool = () => {
                 }
 
                 break;
         }
     };
 
+    /**
+     * Clears the output displayed to the user.
+     */
     const clearOutput = useCallback(() => {
-        setOutput("");
+        setOutput(""); //Memoized function to clear the output.
+        setHasSaved(true);
+        setAllowSave(false);
     }, [setOutput]);
 
+    const handleSaveComplete = useCallback(() => {
+        setHasSaved(true);
+        setAllowSave(false);
+    }, []);
+
     return (
-        <form onSubmit={form.onSubmit((values) => onSubmit({ ...values, tracerouteSwitch: selectedScanOption }))}>
-            <Stack>
-                {UserGuide(title, description_userguide)}
-                <TextInput label={"Hostname/IP address"} {...form.getInputProps("hostname")} />
-                <TextInput label={"Traceroute custom (optional)"} {...form.getInputProps("tracerouteOptions")} />
-                <NativeSelect
-                    value={selectedScanOption}
-                    onChange={(e) => setSelectedTracerouteOption(e.target.value)}
-                    title={"Traceroute option"}
-                    data={tracerouteSwitch}
-                    required
-                    placeholder={"Pick a scan option"}
-                    description={"Type of scan to perform"}
-                />
-                <Button type={"submit"}>start traceroute</Button>
-                {SaveOutputToTextFile(output)}
-                <ConsoleWrapper output={output} clearOutputCallback={clearOutput} />
-            </Stack>
-        </form>
+        <RenderComponent
+            title={title}
+            description={description}
+            steps={steps}
+            tutorial={tutorial}
+            sourceLink={sourceLink}
+        >
+            <form onSubmit={form.onSubmit((values) => onSubmit({ ...values, tracerouteSwitch: selectedScanOption }))}>
+                <Stack>
+                    {UserGuide(title, description)}
+                    <TextInput label={"Hostname/IP address"} {...form.getInputProps("hostname")} />
+                    <TextInput label={"Traceroute custom (optional)"} {...form.getInputProps("tracerouteOptions")} />
+                    <NativeSelect
+                        value={selectedScanOption}
+                        onChange={(e) => setSelectedTracerouteOption(e.target.value)}
+                        title={"Traceroute option"}
+                        data={tracerouteSwitch}
+                        required
+                        placeholder={"Pick a scan option"}
+                        description={"Type of scan to perform"}
+                    />
+                    <Button type={"submit"}>start traceroute</Button>
+                    {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
+                    <ConsoleWrapper output={output} clearOutputCallback={clearOutput} />
+                </Stack>
+            </form>
+        </RenderComponent>
     );
 };
 
```

Comment:
rectify such that tool is reflective only of the actual tool.

Reply:
Could you please explain that a bit more not sure how to implement this.